providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProvider.java

```diff
@@ -173,60 +108,77 @@ public ProviderEvaluation<Value> getObjectEvaluation(
+                    this.startCheckFlagConfigurationChangesDaemon();
         }
         state = ProviderState.READY;
+        super.emitProviderReady(ProviderEventDetails.builder().message("Provider is ready to call the API").build());
         log.info("finishing initializing provider, state: {}", state);
     }
 
+
+    /**
+     * startCheckFlagConfigurationChangesDaemon is a daemon that will check if the flag configuration has changed.
+     *
+     * @return Disposable - the subscription to the observable
+     */
+    @NotNull
+    private Disposable startCheckFlagConfigurationChangesDaemon() {
+        long pollingIntervalMs = options.getFlagChangePollingIntervalMs() != null
+                ? options.getFlagChangePollingIntervalMs() : DEFAULT_POLLING_CONFIG_FLAG_CHANGE_INTERVAL_MS;
+
+        PublishSubject<Object> stopSignal = PublishSubject.create();
+        Observable<Long> intervalObservable = Observable.interval(pollingIntervalMs, TimeUnit.MILLISECONDS);
+        Observable<ConfigurationChange> apiCallObservable = intervalObservable
+                // as soon something is published in stopSignal, the interval will stop
+                .takeUntil(stopSignal)
+                .flatMap(tick -> Observable.fromCallable(() -> this.gofeatureflagController.configurationHasChanged())
+                        .onErrorResumeNext(e -> {
+                            log.error("error while calling flag change API", e);
+                            if (e instanceof ConfigurationChangeEndpointNotFound) {
+                                // emit an item to stop the interval to stop the loop
+                                stopSignal.onNext(new Object());
+                            }
+                            return Observable.empty();
+                        }))
+                .subscribeOn(Schedulers.io());
+
+        Disposable disposable = apiCallObservable
+                .subscribe(
+                        response -> {
+                            if (response == ConfigurationChange.FLAG_CONFIGURATION_UPDATED) {
+                                log.info("clean up the cache because the flag configuration has changed");
+                                this.cacheCtrl.invalidateAll();
+                                super.emitProviderConfigurationChanged(ProviderEventDetails.builder()
+                                        .message("GO Feature Flag Configuration changed, clearing the cache").build());
+                            } else {
+                                log.debug("flag configuration has not changed: {}", response);
+                            }
+                        },
+                        throwable -> log.error("error while calling flag change API, error: {}", throwable.getMessage())
+                );
+
+        // Add a shutdown hook to dispose the subscription on application termination
+        Runtime.getRuntime().addShutdownHook(new Thread(disposable::dispose));
```

Comment:
Provider contains shutdown() method, I think it is more suitable there.

Reply:
It is also called in the `shudown()` method, I can remove this but I kept it for safety reasons.